internal/pkg/manifest/workload.go

```diff
@@ -624,6 +627,59 @@ func (i ImageWithHealthcheck) HealthCheckOpts() *ecs.HealthCheck {
 	return i.HealthCheck.healthCheckOpts()
 }
 
+// PlatformArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type PlatformArgs.
+type PlatformArgsOrString struct {
+	PlatformString *string
+	PlatformArgs   PlatformArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the PlatformArgsOrString
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (p *PlatformArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&p.PlatformArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !p.PlatformArgs.isEmpty() {
+		if err := exec.ValidateOS(p.PlatformArgs.OSFamily); err != nil {
+			return fmt.Errorf("validate OS: %w", err)
+		}
+		if err := exec.ValidateArch(p.PlatformArgs.Arch); err != nil {
+			return fmt.Errorf("validate arch: %w", err)
+		}
```

Comment:
I have a question 🤚🏼  How do we plan to construct the argument for `--platform` from `PlatformArgs` if exactly one of `OS` or `Arch` is `nil`?

I feel like it might make sense to require that both of them are either `nil` or filled at the same time.

Reply:
What do you think of us filling the missing one with the default?

Reply:
Or we can require both or neither. 🤔

Reply:
> Or we can require both

lets do this for the advanced form

Reply:
added validation

Reply:
For the record: 
```
platform:
  osfamily: ''
  architecture: ''
```
are not nil, so they get stopped at the first validation, os.